chain/network/src/network_protocol/proto_conv.rs

```diff
@@ -397,6 +400,7 @@ impl TryFrom<&proto::Block> for Block {
 
 impl From<&PeerMessage> for proto::PeerMessage {
     fn from(x: &PeerMessage) -> Self {
+        let tx_timestamp = SystemTime::now().duration_since(UNIX_EPOCH);
```

Comment:
please don't make time-measurement encoding-dependent and move it to peer.rs.
Here you should just accept a timestamp as an argument.